packages/sitecore-jss/src/graphql/page-list-service.ts

```diff
@@ -0,0 +1,61 @@
+import { SearchQueryService, SearchQueryVariables } from './search-service';
+import { GraphQLClient } from '../graphql-request-client';
+
+// Even though _hasLayout should always be "true" in this query, using a variable is necessary for compatibility with Edge
+const pageListQuery = /* GraphQL */ `
+  query PageListQuery(
+    $rootItemId: String!
+    $language: String!
+    $pageSize: Int = 10
+    $hasLayout: String = "true"
+    $after: String
+  ) {
+    search(
+      where: {
+        AND: [
+          { name: "_path", value: $rootItemId, operator: CONTAINS }
+          { name: "_language", value: $language }
+          { name: "_hasLayout", value: $hasLayout }
+        ]
+      }
+      first: $pageSize
+      after: $after
+    ) {
+      total
+      pageInfo {
+        endCursor
+        hasNext
+      }
+      results {
+        url {
+          path
+        }
+      }
+    }
+  }
+`;
+
+/**
+ * The schema of data returned in response to a page list query request
+ */
+export type PageListQueryResult = { url: { path: string } };
+
+/**
+ * Service that fetches page list data using Sitecore's GraphQL API.
+ * @augments SearchQueryService<PageListQueryResult>
+ */
+export class PageListService extends SearchQueryService<PageListQueryResult> {
```

Comment:
This `PageListService` seems fairly specific to the single usage in sitemap service. Given that we need to customize the query in sitemap service (see other comment), not sure this abstraction is providing much?

Reply:
What about generating sitemap (like, the sitemap that crawlers use). Or navigation.

Reply:
As agreed upon, the extra PageListService abstraction has been removed